app/blog/page.tsx

```diff
@@ -0,0 +1,81 @@
+import { Data } from "@measured/puck";
+import { Render } from "@measured/puck/rsc";
+import { Metadata } from "next";
+import { notFound } from "next/navigation";
+
+import content from "../../content.json";
+import { getPosts } from "../../lib/get-posts";
+import { supabase } from "../../lib/supabase";
+import config from "../../puck.config";
+
+const { blogDescription, openGraphLocale, siteUrl, title } = content;
+
+const getPageRes = (path: string) =>
+  supabase.from("puck").select("*").eq("path", path).maybeSingle();
+
+export { viewport } from "../page";
+
+export async function generateMetadata(): Promise<Metadata> {
+  const path = "/blog";
+  const pageRes = await getPageRes(path);
+  const data = pageRes?.data?.data as Data;
```

Comment:
Seems like a future opportunity for a refactor as it's now used in a few places

```tsx
  const { data } = await getPageData("/blog");
```

Reply:
Fair